### ISSUE TITLE
PostgreSQL client interface errors

### ISSUE THREAD

**Issue by MarcoTrevisiol** [questdb/questdb#5202]
### To reproduce

To create and init a sample table:
```sql
CREATE TABLE bugtest(
  timestamp TIMESTAMP,
  symbol SYMBOL,
  price DOUBLE,
  amount DOUBLE
  ) TIMESTAMP(timestamp)
PARTITION BY DAY;

INSERT INTO bugtest
VALUES
    ('2021-10-05T11:31:35.878Z', 'AAPL', 245, 123.4),
    ('2021-10-05T12:31:35.878Z', 'AAPL', 245, 123.3),
    ('2021-10-05T13:31:35.878Z', 'AAPL', 250, 123.1),
    ('2021-10-05T14:31:35.878Z', 'AAPL', 250, 123.0);
```

The problem appears sometime, randomly, when query from a client using PostgreSQL driver (exposed at port 8812 in the docker image). Sometimes I get no data or seemingly corrupted data (at least with more elaborate queries), the other times I get the correct answer.

I tried CLI tool "psql" from an ubuntu machine many times:
```bash
while :
do
    psql -d "$ConnectionString" -c "SELECT 1 FROM bugtest"
done
```
I also tried python driver from a windows machine (conda installation) with the following script:
```python
# script.py
import os
import psycopg2

quest_connection_string = os.environ.get("ConnectionString")
query = "SELECT 1 FROM bugtest"

with psycopg2.connect(quest_connection_string) as conn:
    with conn.cursor() as cursor:
        cursor.execute(query)
        response = cursor.fetchall()
        print(response)
```
again, the effect is visible only sometimes, therefore I do:
```bash
while :
do
    python script.py
done
```

### Additional info

- The query `SELECT 1 FROM bugtest` seems the simplest I could get to fail. Strangely, if I query `SELECT 'x' FROM bugtest` I never found any weirdness.
- The web interface, the one exposed at port 9000 in the docker image, never seems to give any weirdness.
- The log of the server does not register any error. It even cache the query regularly without anything strange.
- I think I got this error with the latest questDB version, although I am not able to prove it did not existed before.

Could this be a bug of the questDB server?

### QuestDB version:

8.2.0

### OS, in case of Docker specify Docker and the Host OS:

Ubuntu 24.04 (docker)

### File System, in case of Docker specify Host File System:

zfs

### Full Name:

Marco Trevisiol

### Affiliation:

Deimos

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [X] Yes, I have

**Comment by enricobenedos** [comment]
@mtopolnik, resuming from Slack, we tried with a cleaned instance without using ZFS and we aren't able to reproduce it. 
System wasn't in idle but it was with low load. Configuration is the default one. But our production instance have been updated more times since the installation.

**Comment by mtopolnik** [comment]
We solved a number of bugs related to PGWire since the release of 8.2.0. I suggest you upgrade to that new version once it's out, and see if it occurs again. If it does, the fast remedy to try is the configuration option `pg.legacy.mode.enabled=true`. It would be of immense value to know if this config change removed the problem, since it would narrow down its origin to the new PGWire code.

**Comment by bluestreak01** [comment]
i can repro the issue, will keep you posted

**Comment by enricobenedos** [comment]
@mtopolnik did you mean something different from `8.2.0`? Because that's exactly the one we were using with the bug. Downgrading it fixed with no issue. Are there incoming new releases?

@bluestreak01 it should be perfect

**Comment by enricobenedos** [comment]
Unfortunately we lost them once removed and restarted with the old version.

**Comment by enricobenedos** [comment]
But this morning we update another time to `8.2.0` and we are able to reproduce it.

The log since start of new `8.2.0` container: [questdb_error.log](https://github.com/user-attachments/files/17944215/questdb_error.log)

By us there aren't logs that show something wrong but the problem exists.

**Comment by mtopolnik** [comment]
@enricobenedos Yes, I meant the upcoming release that includes several bugfixes related to PGWire.

>  this morning we update another time to 8.2.0 and we are able to reproduce it.

Can you please apply the config option `pg.legacy.mode.enabled=true` to see if it goes away? Thank you for the logs!

**Comment by MarcoTrevisiol** [comment]
This is a screenshot of what I am experiencing today
![image](https://github.com/user-attachments/assets/4be81f3b-e603-443f-85ea-1be33eca2725)

Strangely, it seems like the frequency of this weird behaviour has decrease since yesterday, so that I took me several attempts to record one incident.

**Comment by enricobenedos** [comment]
@mtopolnik we applied that. Now we need some testing time.

**Comment by MarcoTrevisiol** [comment]
@mtopolnik After turning on that config option and some test, we experienced nothing strange.

**Comment by ideoma** [comment]
Hey @MarcoTrevisiol we think the problem is fixed in https://github.com/questdb/questdb/pull/5189. 

Can you please try running your test on the nightly docker build
https://registry.hub.docker.com/layers/questdb/questdb/nightly/images/sha256-a67d47479213f1c38614a0279adcfb49db89be6b1252fa9fdce6f69cc0f315c4?context=explore
?

We appreciate your help.

**Comment by mtopolnik** [comment]
I'd add one potential gotche when testing: be sure to reset the config to `pg.legacy.mode.enabled=false`

**Comment by enricobenedos** [comment]
Hello @ideoma, we are working with only one **production** instance. We don't know if it is a good idea to try a nightly build.

**Comment by mtopolnik** [comment]
We managed to reproduce this on 8.2.0, using a combination of JDBC and `psql`. It did not reproduce on our current master.

**Comment by Dapeng960208** [comment]
When I was using version 8.2, the following error message would appear during the query. I have confirmed that there is no problem with the query statement, but the problem is accidental, and the error message is as follows:

java.util.concurrent.atomic.AtomicBoolean.get()" because "this.cancelledFlag" is null

**Comment by mtopolnik** [comment]
Hi @Dapeng960208, this sounds like an unrelated issue. Can you please create a new one and add as much detail as you have?

**Comment by MarcoTrevisiol** [comment]
> We managed to reproduce this on 8.2.0, using a combination of JDBC and `psql`. It did not reproduce on our current master.

Indeed we found no more issues since that fix (version 8.2.1).
